docs/compatibility.md

```diff
@@ -46,6 +46,93 @@ guarantees, but please let us know if this is an important use case for you,
+or directly in C++ or Python APIs. Serialization needs a target version of
+StableHLO to write an artifact written in `#.#.#` format (See [Version.h](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/Version.h#:~:text=getCurrentVersion)
+for current version). Deserialization uses the current version of StableHLO to
+read an artifact.
+
+### Using the `stablehlo-translate` tool
+
+This is the easiest way to create and read a portable artifact.
+
+```bash
+# Write a StableHLO program to a portable artifact
+$ stablehlo-translate --serialize file.mlir --target=0.9.0 > portable_artifact.mlir.bc
+
+# Read StableHLO portable artifact
+$ stablehlo-translate --deserialize portable_artifact.mlir.bc
+```
+
+### Using C++ APIs
+
+For programmatic workflows, StableHLO provides the following APIs to create
+portable artifacts:
+
+```c++
+// From: #include "stablehlo/api/PortableApi.h"
+
+// Get the current StableHLO version.
+//
+// This value can be used as the `targetVersion` argument to
+// `serializePortableArtifact`.
+std::string getCurrentVersion();
+
+// Get the minimum supported StableHLO version.
+//
+// This value can be used as the `targetVersion` argument to
+// `serializePortableArtifact`.
+std::string getMinimumVersion();
+
+// From: #include "stablehlo/dialect/Serialization.h"
+
+// Write a StableHLO program to a portable artifact
+// Writes a stable payload for `module` to `os`. If compatibility with a
+// previous version of StableHLO is required, provide the required version
+// string `#.#.#` for `targetVersion`.
+//
+// Can fail if `module` cannot be expressed in the `targetVersion` version of
+// StableHLO, e.g. if it's using new or removed features, or if it involves
+// unsupported dialects.
+LogicalResult serializePortableArtifact(ModuleOp module,
+                                        StringRef targetVersion,
+                                        raw_ostream& os);
+
+// Read StableHLO portable artifact
+//
+// Can fail if `sourceStr` cannot be expressed in the current version of
+// StableHLO, e.g. if it's using incompatible features. Returns nullptr if
+// `sourceStr` is invalid or fails to deserialize.
+OwningOpRef<ModuleOp> deserializePortableArtifact(StringRef sourceStr,
+                                                  MLIRContext* context);
+```
+
+See [`stablehlo/api/PortableApi.h`](https://github.com/openxla/stablehlo/blob/main/stablehlo/api/PortableApi.h)
+and [`stablehlo/dialect/Serialization.h`](https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/Serialization.h)
+for full APIs.
+
+See [`StablehloTranslateMain.cpp`](https://github.com/openxla/stablehlo/blob/main/stablehlo/tools/StablehloTranslateMain.cpp#:~:text=serializePortableArtifact)
+for example usage of these APIs.
+
+### Using Python APIs
+
+StableHLO also provides Python bindings to the C++ Serializaiton APIs:
+
+```python
+version = stablehlo.get_current_version()
+version = stablehlo.get_minimum_version()
```

Comment:
This is a bit confusing. How about `# or alternatively: version = stablehlo.get_minimum_version()`, or something along these lines?

Reply:
Used pyi format to describe APIs